tests/MSBuildDeviceIntegration/Tests/XASdkDeployTests.cs

```diff
@@ -156,6 +157,70 @@ public void DotNetDebug ()
 			WaitFor (2000);
 			Assert.IsTrue (breakpointHit, "Should have a breakpoint");
 		}
+
+		[Test]
+		public void TypeAndMemberRemapping ([Values (false, true)] bool isRelease)
+		{
+			AssertHasDevices ();
+
+			var proj = new XASdkProject () {
+				IsRelease = isRelease,
+				OtherBuildItems = {
+					new AndroidItem._AndroidRemapMembers ("remap.xml") {
+						TextContent = () => @"<?xml version=""1.0"" encoding=""utf-8"" ?>
+<replacements>
+  <replace-type from='android/app/Activity' to='example/RemapActivity' />
+  <replace-method
+      source-type='example/RemapActivity'
+      source-method-name='onCreate'
+      target-type='example/RemapActivity'
+      target-method-name='onMyCreate' target-method-instance-to-static='false' />
+  <replace-method
+      source-type='android/view/View'
+      source-method-name='setOnClickListener'
+      target-type='example/ViewHelper'
+      target-method-name='mySetOnClickListener' target-method-instance-to-static='true' />
+</replacements>
+",
+					},
+					new AndroidItem.AndroidJavaSource ("RemapActivity.java") {
+						Encoding = Encoding.UTF8,
+						TextContent = () => ResourceData.RemapActivity,
+						Metadata = {
+							{ "Bind", "True" },
+						},
+					},
+				},
+			};
+			proj.MainActivity = proj.DefaultMainActivity.Replace (": Activity", "global::Example.RemapActivity");
+			proj.SetRuntimeIdentifier (DeviceAbi);
+			var relativeProjDir = Path.Combine ("temp", TestName);
+			var fullProjDir     = Path.Combine (Root, relativeProjDir);
+			TestOutputDirectories [TestContext.CurrentContext.Test.ID] = fullProjDir;
+			var files = proj.Save ();
+			proj.Populate (relativeProjDir, files);
+			proj.CopyNuGetConfig (relativeProjDir);
+			var dotnet = new DotNetCLI (proj, Path.Combine (fullProjDir, proj.ProjectFilePath));
+
+			Assert.IsTrue (dotnet.Build (), "`dotnet build` should succeed");
```

Comment:
The other `XASdkTests` class has some helper methods:

https://github.com/xamarin/xamarin-android/blob/1efa0cf46c9079fc06669a4434f597faf47504af/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/XASdkTests.cs#L1071-L1094

Should we put these somewhere, so that `XASdkDeployTests` can use them?

Then you can do:
```csharp
var builder = CreateDotNetBuilder (proj);
Assert.IsTrue (builder.Build (), $`dotnet build` should succeed");
```

Reply:
This test looks good. But making use of methods like `CreateDotNetBuilder` would make it a bit nicer :)